### PR TITLE
SLCORE-237 Add support of SonarGo 1.6 (slang-based)

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by m-g-sonar** [comment]
@henryju I have not idea why ITs does not run, but it fails also on master for me. I also tried branch #254 to check if it could come from this, but I get the same issue.

Note that without having at least this PR passing QA, I can not move forward with the SLang ITs checking for compatiblity with SonarGo. https://github.com/SonarSource/slang-enterprise/pull/86

File: core/src/main/resources/plugins_min_versions.txt [hunk 1]
```diff
@@ -19,5 +19,6 @@ tsql=1.1.0.2177
 typescript=1.5.0.2122
 web=3.0.0.1411
 xml=2.0.1.2020
+go=1.6.0.673
```
Comment (henryju) [inline core/src/main/resources/plugins_min_versions.txt]: Can you please keep alphabetic ordering?

File: its/tests/src/test/java/its/ConnectedModeTest.java [hunk 2]
```diff
@@ -114,6 +115,7 @@
     .addPlugin(MavenLocation.of("org.sonarsource.slang", "sonar-kotlin-plugin", System.getProperty("kotlinVersion")))
     .addPlugin(MavenLocation.of("org.sonarsource.slang", "sonar-ruby-plugin", System.getProperty("rubyVersion")))
     .addPlugin(MavenLocation.of("org.sonarsource.slang", "sonar-scala-plugin", System.getProperty("scalaVersion")))
+    .addPlugin(MavenLocation.of("org.sonarsource.slang", "sonar-go-plugin", System.getProperty("goVersion")))
```
Comment (henryju) [inline its/tests/src/test/java/its/ConnectedModeTest.java]: For the system property to exist, you have to update also https://github.com/SonarSource/sonarlint-core/blob/master/run-integration-tests.sh

File: core/src/main/resources/plugins_stream_support.txt [hunk 3]
```diff
@@ -16,4 +16,5 @@ kotlin=1.1
 ruby=1.1
 sonarscala=1.4
 tsql=1.1
-xml=2.0.1
\ No newline at end of file
+xml=2.0.1
+go=1.6
```
Comment (m-g-sonar) [inline core/src/main/resources/plugins_stream_support.txt]: just realized that there is two times XML in this file.

File: run-integration-tests.sh [hunk 4]
```diff
@@ -37,6 +38,8 @@ case "$SQ_VERSION" in
     KOTLIN_VERSION=LATEST_RELEASE
     RUBY_VERSION=LATEST_RELEASE
     SCALA_VERSION=LATEST_RELEASE
+    # SonarLint-compatible Go plugin not yet released
+    GO_VERSION=DEV
```
Comment (m-g-sonar) [inline run-integration-tests.sh]: to be changed once released?

**Comment by m-g-sonar** [comment]
Updated according to what we discussed @henryju ! Thanks for the support!

File: core/src/main/resources/plugins_stream_support.txt (hunk 3 above)
Comment (henryju) [reply core/src/main/resources/plugins_stream_support.txt]: That's the benefit of lexicographic sorting like in the other file ;)

File: run-integration-tests.sh (hunk 4 above)
Comment (m-g-sonar) [reply run-integration-tests.sh]: Yep that's why I changed the version to a fixed version!

**Comment by m-g-sonar** [comment]
As discussed offline, we postpone the support of SonarLint for SonarG, due to performance investigation.

**Pull request closed without merging**